mono/metadata/threads.c

```diff
@@ -1740,39 +1740,56 @@ ves_icall_System_Threading_Thread_SetPriority (MonoThreadObjectHandle this_obj,
 
 /* If the array is already in the requested domain, we just return it,
    otherwise we return a copy in that domain. */
-static MonoArray*
-byte_array_to_domain (MonoArray *arr, MonoDomain *domain, MonoError *error)
+static MonoArrayHandle
+byte_array_to_domain (MonoArrayHandle arr, MonoDomain *domain, MonoError *error)
 {
-	MonoArray *copy;
+	HANDLE_FUNCTION_ENTER ()
 
-	error_init (error);
-	if (!arr)
-		return NULL;
+	if (!arr || MONO_HANDLE_IS_NULL (arr))
+		return MONO_HANDLE_NEW (MonoArray, NULL);
 
-	if (mono_object_domain (arr) == domain)
+	if (MONO_HANDLE_DOMAIN (arr) == domain)
 		return arr;
 
-	copy = mono_array_new_checked (domain, mono_defaults.byte_class, arr->max_length, error);
-	memmove (mono_array_addr (copy, guint8, 0), mono_array_addr (arr, guint8, 0), arr->max_length);
-	return copy;
+	size_t const size = mono_array_handle_length (arr);
+
+	// Capture arrays into array for looping.
```

Comment:
Couldn't you just use 2 local variables here? The loop really obfuscate what you are trying to do for no apparent reason, there are only 2 elements and AFAICT there isn't going to be a third one.

Reply:
2 variables, and then 2 pins, and 2 unpins, prefer that? It seems clear with loops.

Reply:
It isn't clear to me who didn't write the code :wink:

Reply:
Even with the comment?